### PR TITLE
Fix modification of source file during gdal provider test

### LINKED ISSUE #48846: test_core_gdalprovider modifies tests/testdata/zip/landsat_b1.zip source file
### What is the bug or the crash?

Spin-off of #25830 specific for a single test `test_core_gdalprovider`.

### Steps to reproduce the issue

1. build
2. git checkout . # ensure no local modification exist
3. builddir$ xvfb-run ctest --output-on-failure -R test_core_gdalprovider
4. srcdir$ git status -uno # will show tests/testdata/zip/landsat_b1.zip being modified

### Versions

Current master branch as of June 1st 2022 - 1af9c8150c0c45b1bd0072c5e707c45cdfb88508

### Supported QGIS version

- [x] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

Additional context can be found in https://github.com/qgis/QGIS/pull/48654#issuecomment-1145762759 where I've added automatic check of source tree NOT being changed (reviews welcome).

### PR DESCRIPTION
**Pull request by strk**
Disables stats creation during the run of the test
Closes GH-48846
References GH-25830

### CODE REVIEW

**Comment by strk** [comment]
This may or may not be the only test opening that zip file. Running the WHOLE testsuite under CI conditions would be useful to check this, but it's a bit harder until #48752 is merged in (will allow calling `.ci/run_tests.sh` from worktrees)

File: tests/src/core/testqgsgdalprovider.cpp [hunk 1]
```diff
@@ -86,6 +87,11 @@ void TestQgsGdalProvider::initTestCase()
 
   mSupportsNetCDF = GDALGetDriverByName( "netcdf" ) != NULL;
 
+  // Disable creation of .aux.xml (stats) files during test run,
+  // to avoid modifying .zip files.
+  // See https://github.com/qgis/QGIS/issues/48846
+  setenv( "GDAL_PAM_ENABLED", "NO", 1 );
```
Comment (rouault) [inline tests/src/core/testqgsgdalprovider.cpp:93]: setenv() is not portable:
```suggestion
  CPLSetConfigOption( "GDAL_PAM_ENABLED", "NO" );
```

Comment (strk) [reply tests/src/core/testqgsgdalprovider.cpp:93]: Thanks ! Interesting the same setting is already found in other tests..

**Review by rouault — APPROVED** [review]

**Comment by strk** [comment]
The backport to 3.22 is in #48926 -- should I remove the backport-related labels from here ?

**Pull request merged**